moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js

```diff
@@ -174,14 +178,21 @@ const ProjectWorkActivitiesTable = () => {
 
   const onDeleteActivity = useCallback(
     ({ id }) => {
-      window.confirm("Are you sure you want to delete this activity?") &&
-        deleteContract({ variables: { id } }).then(() => {
-          refetch();
-        });
+      setActivityToDelete(id);
+      setIsDeleteConfirmationOpen(true);
     },
-    [deleteContract, refetch]
+    []
   );
 
+  const handleDeleteConfirmed = useCallback(() => {
+    if (activityToDelete) {
+      deleteContract({ variables: { id: activityToDelete } }).then(() => {
+        refetch();
+        setActivityToDelete(null);
+      });
```

Comment:
i noticed that down in the `onSubmitCallback` we use the `refetch().then(...)` pattern, and it had me wondering if we should use that here, so that we would wait to `setActivityToDelete()` until the refetch completes, but i do not really see any benefit in that. 

anyway, because `refetch()` returns a `promise` you should be able to chain it like this, if you wanted to:

```javascript
      deleteContract({ variables: { id: activityToDelete } })
        .then(() => refetch())
        .then(() => setActivityToDelete(null));
```

and—this is just an observation—looking at the DeleteConfirmationModal [code](https://github.com/cityofaustin/atd-moped/blob/49feca21a2b80ce53d5113735dd08b01725d8f6d/moped-editor/src/views/projects/projectView/DeleteConfirmationModal.js#L44-L47), it would be a nice future enhancement to make the `handleDeleteClose` callback wait until the `submitDelete` function completes—that would prevent the modal from closing until the mutation actually completes.

that's totally outside the scope of this work—my mind just went there when i was looking at the callback behavior in this PR.

Reply:
great catch - @johnclary! i'll create an spinoff snackoo issue to fix up `DeleteConfirmationModal` in this way later on. I think that will be a great enhancement too. 🙏